hivemq-edge/src/main/java/com/hivemq/api/resources/impl/ProtocolAdaptersResourceImpl.java

```diff
@@ -376,10 +376,8 @@ protected void validateAdapterSchema(
             return;
         }
 
-        final ProtocolAdapterFactory<?> protocolAdapterFactory =
-                protocolAdapterManager.getProtocolAdapterFactory(information.getProtocolId());
         final ProtocolAdapterSchemaManager protocolAdapterSchemaManager =
-                new ProtocolAdapterSchemaManager(objectMapper, protocolAdapterFactory.getConfigClass());
+                new ProtocolAdapterSchemaManager(objectMapper, protocolAdapterWritingService.writingEnabled() ? information.configurationClassWriting() : information.configurationClassReading());
```

Comment:
confgurationClassWriting implies that it is the config class for writing, but actually it is the config class for both reading and writing